t/lib/Test/Deep/JSON.pm

```diff
@@ -0,0 +1,42 @@
+use strict;
+use warnings;
+package Test::Deep::JSON;
+
+use Test::Deep ();
+
+use Exporter 'import';
+our @EXPORT = qw( jcmp_deeply jstr jnum jtrue jfalse );
+
+sub jcmp_deeply {
+  local $Test::Builder::Level = $Test::Builder::Level + 1;
+  local $Test::Deep::LeafWrapper = \&Test::Deep::str;
+  Test::Deep::cmp_deeply(@_);
+}
+
+sub jstr  { Test::Deep::all( Test::Deep::obj_isa('JSON::Typist::String'),
+                             Test::Deep::str($_[0])) }
+
+sub jnum  { Test::Deep::all( Test::Deep::obj_isa('JSON::Typist::Number'),
+                             Test::Deep::num($_[0])) }
+
+sub jtrue {
```

Comment:
Maybe a 'jbool' that these two use internally, gives users way to simply check type of something without caring if true/false..

    cmp_deeplly($thing->{field}, jbool())